packages/web-components/src/components/va-search-input/va-search-input.tsx

```diff
@@ -0,0 +1,402 @@
+   */
+  @Prop() buttonText?: string;
+
+  /**
+   * The aria-label for search input and button. Default is 'Search'.
+   */
+  @Prop() label?: string = 'Search';
+
+  /**
+   * An array of strings containing suggestions to be displayed in listbox.
+   * This component displays up to 5 suggestions.
+   */
+  @Prop() suggestions?: any;
+
+  /**
+   * The value of the input field
+   */
+  @Prop() value?: string = '';
+
+  /**
+   * If suggestions are provided, then format suggestions and open the listbox.
+   * Limits suggestions to 5 and sorts them.
+   */
+  componentDidLoad() {
+    if (!Array.isArray(this.suggestions) || !this.suggestions?.length) return;
+    this.updateSuggestions(this.suggestions);
+  }
+
+  /**
+   * If suggestions are provided, then format suggestions and open the listbox.
+   * Limits suggestions to 5 and sorts them.
+   */
+  @Watch('suggestions')
+  watchSuggestionsHandler(newSuggestions: string[]) {
+    if (!Array.isArray(newSuggestions)) return;
+    this.updateSuggestions(newSuggestions);
+  }
+
+  // Host event handlers
+  /**
+   * Closes listbox when focus is outside of the host element
+   */
+  private handleBlur = () => {
+    if (!this.formattedSuggestions.length) return;
+    this.isListboxOpen = false;
+  };
+
+  /**
+   * Fires a submit event
+   */
+  private handleSubmit = () => {
+    this.el.dispatchEvent(
+      new CustomEvent('submit', {
+        bubbles: true,
+        cancelable: true,
+        composed: true,
+        detail: { value: this.inputRef.value },
+      }),
+    );
+  };
+
+  // Input event handlers
+  /**
+   * Updates suggestion formatting as user types
+   */
+  private handleInput = () => {
+    if (!this.suggestions) return;
+    this.updateSuggestions(this.suggestions);
+  };
+
+  /**
+   * Opens listbox when search input has focus and suggestions are provided
+   */
+  private handleInputFocus = () => {
+    if (this.formattedSuggestions.length && !this.isListboxOpen) {
+      this.isListboxOpen = true;
+    }
+  };
+
+  /**
+   * Implements keyboard interface from Keyboard Support at
+   * https://www.w3.org/TR/wai-aria-practices-1.1/examples/combobox/aria1.1pattern/listbox-combo.html
+   *
+   * Enter key was added to fire a submit event.
+   * Tab key was added to aid in isListboxOpen state management.
+   */
+  private handleInputKeyDown = (event: KeyboardEvent) => {
+    const options = this.el.shadowRoot.querySelectorAll(
+      '[role="option"]',
+    ) as NodeListOf<HTMLLIElement>;
+    const firstOption = options[0];
+    const lastOption = options[options.length - 1];
+
+    switch (event.key) {
+      case 'ArrowDown':
+        // Option doesn't exist if suggestions aren't provided
+        if (!firstOption) return;
+        this.selectSuggestion(firstOption);
+        break;
+      case 'ArrowUp':
+        // Option doesn't exist if suggestions aren't provided
+        if (!lastOption) return;
+        this.selectSuggestion(lastOption);
```

Comment:
Would these if statements work the same if they were checking:

```js
if (!options) return;
```
Having them selectively look at the first or the last option is a bit confusing to me because I'm not understanding why those particular items are important.

Reply:
You're correct. It doesn't matter if there's a first or last option - what matters is that there's at least 1 option available.